File: easy/php/cmd_exec/InterfaceImpl.php
<?php

require_once __DIR__ . '/../common/VulnChild.php';
require_once __DIR__ . '/../iface/VulnRunner.php';

class InterfaceImpl extends VulnChild implements VulnRunner
{
    public function run(string $cmd): string
    {
        return $this->exec($cmd);
    }
}

File: easy/rust/cmd_exec/interface_impl.rs
use crate::common::vuln_child::VulnChild;
use crate::iface::vuln_runner::VulnRunner;

pub struct InterfaceImpl {
    pub child: VulnChild,
}

impl VulnRunner for InterfaceImpl {
    fn run(&self, cmd: &str) {
        self.child.parent.exec(cmd);
    }
}
